### ISSUE TITLE
Xcode install used to take a few seconds, now takes ~6 minutes

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#103232]
Looking at the Mac clang-tidy builder: https://ci.chromium.org/p/flutter/builders/prod/Mac%20clang-tidy

Somewhere around 4/27, the Xcode install step went from regularly taking a few seconds:

https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20clang-tidy/1068/overview

To consistently taking >6 minutes:

https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20clang-tidy/1069/overview

The switch seems to have happened *before* impeller came in on the 27th.

Other mac builds also appear to be affected.

**Comment by CaseyHillers** [comment]
This is related to my change in https://flutter-review.googlesource.com/c/infra/+/28063

We moved off of hand coding the caches in the yaml to basing it on the dependencies list.

This is another dependency missed as it's not explicitly in the `dependencies` list. I'll send a CL to add it.

**Comment by CaseyHillers** [comment]
The proposed fix is:

```starlark
    if xcode:
      name = repo + "_" + version + "_xcode_" + xcode
      swarming_caches.append(swarming.cache(name = name + "_legacy", path = "osx_sdk"))
      swarming_caches.append(swarming.cache(name = name, path = "xcode"))
```

However, that's getting the error:

```
Error: bad "caches": caches swarming.cache(name = "flutter_2_13_0_xcode_13a233_legacy", path = "osx_sdk", wait_for_warm_cache = None) and swarming.cache(name = "flutter_2_13_0_xcode_default_legacy", path = "osx_sdk", wait_for_warm_cache = None) use same path
```

I'll need to debug this some more.

**Comment by CaseyHillers** [comment]
I see the issue, flutter/flutter has a special format for this as well:

```
      dependencies: >-
        [
          {"dependency": "xcode"},
          {"dependency": "gems"},
          {"dependency": "apple_signing"}
        ]
      os: Mac-12
      cpu: x86
      device_os: iOS-15.1
      xcode: 13a233
```

@jmagman do you know why we have xcode listed as a property and a dependency? Should the dependency be ignored?

**Comment by jmagman** [comment]
I have no idea, I think you set it up that way? #87694

**Comment by CaseyHillers** [comment]
![image](https://user-images.githubusercontent.com/2148558/167226783-60ab000f-769e-4aac-9822-bfa9c6eb0cbc.png)

Thanks for verifying, https://flutter-review.googlesource.com/c/infra/+/29683 should fix this. I can follow up to remove the dependency

**Comment by CaseyHillers** [comment]
I manually triaged a few Mac builds and they seem to be back to a few seconds.

Feel free to reopen if you're seeing instances of it not using the cache.

**Comment by zanderso** [comment]
Not sure when the change landed, but here are a couple instances from presubmit checks not hitting the cache after your comment:

https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20clang-tidy/3182/overview
https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20Web%20Engine/26329/overview

**Comment by yusuf-goog** [comment]
@CaseyHillers  Looks like install xcode had come down, but just went back up: 

https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20clang-tidy/1707/overview 
https://ci.chromium.org/ui/p/flutter/builders/prod/Mac%20clang-tidy/1708/overview

similarly for other builds as well.

**Comment by keyonghan** [comment]
> Not sure when the change landed, but here are a couple instances from presubmit checks not hitting the cache after your comment:

> https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20clang-tidy/3182/overview
> https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20Web%20Engine/26329/overview

Taking the first build https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20clang-tidy/3182/overview for example, it ran after the CL fixes. It ran on bot https://chromium-swarm.appspot.com/bot?id=build850-m9, and build 3182 is the first time it ran an `engine` builder. This is expected as now the cache is repo specific. Will keep this bug open to see if the following engine builds can see the cache.

**Comment by keyonghan** [comment]
Yeah, validated the second build: https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20Web%20Engine/26329/overview. It ran on bot https://chromium-swarm.appspot.com/bot?id=build873-m9 as the first `engine` build. The following second engine build on the same bot picked up the cache:
https://luci-milo.appspot.com/ui/p/flutter/builders/try/Mac%20Web%20Engine/26334/overview

**Comment by keyonghan** [comment]
Closing. Please reopen if any other issues.
